bdk_chain/src/tx_graph.rs

```diff
@@ -167,38 +175,84 @@ impl TxGraph {
         })
     }
 
-    /// Iterate over all full transactions in the graph
+    /// Iterate over all full transactions in the graph.
     pub fn full_transactions(&self) -> impl Iterator<Item = &Transaction> {
         self.txs.iter().filter_map(|(_, tx)| match tx {
             TxNode::Whole(tx) => Some(tx),
             TxNode::Partial(_) => None,
         })
     }
 
+    /// Iterate over all partial transactions (outputs only) in the graph.
     pub fn partial_transactions(&self) -> impl Iterator<Item = (Txid, &BTreeMap<u32, TxOut>)> {
         self.txs.iter().filter_map(|(txid, tx)| match tx {
             TxNode::Whole(_) => None,
             TxNode::Partial(partial) => Some((*txid, partial)),
         })
     }
 
-    /// Return an iterator of conflicting txids, where the first field of the tuple is the vin of
-    /// the original tx in which the txid conflicts.
-    pub fn conflicting_txids<'g>(
+    /// Iterates descendant transactions of given `txid` (including the given `txid`).
+    ///
+    /// if `path_filter` returns `true`, we will continue exploring the path from the `txid` of
+    /// `depth`, where `depth: usize` is the distance from the initial `txid` to the descendant.
+    pub fn iter_tx_descendants<'g>(
+        &'g self,
+        txid: Txid,
+        path_filter: impl FnMut(usize, Txid) -> bool + 'g,
+    ) -> impl Iterator<Item = (usize, Txid)> + '_ {
```

Comment:
Whether we know the full descendant graph or not is not relevant. `TxGraph` will contain conflicting spends within itself. The idea of `path_filter` is we can start from a `txid` and ignore paths that are irrelevant to us.

Reply:
Currently `path_filter` maybe sub-optimal, because we filter paths individually, instead of choosing between "conflicting paths", which can help us construct consistent histories.

Reply:
Okay.. It's still not fully clear to me how a user can ever provide a path filter without analyzing the full graph himself or by making some other conflict identification code with the full graph data.. But looks okay for now.. I suppose in most part of the code we are gonna use `no_filter` anyway..

Reply:
> It's still not fully clear to me how a user can ever provide a path filter without analyzing the full graph himself 

One way is using the sparsechain.

I agree that the implementation is sub-optimal. Open for ideas.